### ISSUE TITLE
question re: Elixir compatibility

### ISSUE THREAD

**Issue by cameronjacobson** [benoitc/couchbeam#131]
I haven't done thorough testing, but it appears this library just runs in the Elixir environment.  Without any customization.  Are there any plans to integrate with Elixir ecosystem or test for compatibility with that language?

**Comment by optikfluffel** [comment]
Elixir by design can use Erlang code and vice versa, 'cause both boils down to BEAM byte code, there isn't really anything that needs to be tested for this.

> Elixir compiles into BEAM byte code (via Erlang Abstract Format). This means that Elixir code can be called from Erlang and vice versa, without the need to write any bindings.
> (from [Erlang/Elixir Syntax: A Crash Course](http://elixir-lang.org/crash-course.html#adding-elixir-to-existing-erlang-programs))

But it would be cool if couchbeam would be [published to Hex](https://hex.pm/docs/rebar3_publish).

**Comment by cameronjacobson** [comment]
Thank you for the reply.  I was looking for an authoritative source on that.  The other night I was reading and saw that there was a less confident explanation.  Something along the lines of what you're saying, but with the caveat "with few exceptions".  It sounds like the crash course is 100% definitive, and can't get a more authoritative source than the elixir site.  It must be that the "exceptions" part has been resolved, or the source I was reading from was misinformed.

**Comment by benoitc** [comment]
@cameronjacobson which exceptions? I am not aware of any. So far couchbeam is based on hackney witch 100% works on elixir. So it should be OK. If you see anything just let me know and I will fix it ASAP.

**Comment by cameronjacobson** [comment]
@benoitc Thank you for the reply.  I spent a little time trying to find the quote I talked about above, but was unsuccessful.  I think it doesn't matter now though since a comment in this thread from @optikfluffel points to a definitive answer from a reputable source.

The other thing, which I implied (should've stated specifically) but optikfluffel detailed, was about Hex.pm.  So I'll leave the ticket open.  If you don't intend to integrate w/ Hex, or would prefer to open a separate ticket specific to that issue, this ticket could be closed.
